src/wasm-s-parser.h

```diff
@@ -205,7 +205,7 @@ class SExpressionParser {
       input++;
       return allocator.alloc<Element>()->setString(IString(str.c_str(), false), dollared);
     }
-    while (input[0] && !isspace(input[0]) && input[0] != ')') input++;
+    while (input[0] && !isspace(input[0]) && input[0] != ')' && input[0] != '(') input++;
```

Comment:
would be good to split that out into a bugfix with a testcase, then. but surely this would already be tested?